external-crates/move/crates/move-vm-runtime/src/execution/tracing/tracer.rs

```diff
@@ -320,7 +320,9 @@ impl VMTracer<'_> {
         machine: &MachineState,
         stack_idx: usize,
     ) -> Option<TraceValue> {
+        // TODO: Add an assertion establishing that operand_stack and type_stack are in sync.
         if stack_idx >= machine.operand_stack.len() {
+            // TODO: Should we emit a trace error here?
```

Comment:
We actually do! If you look at the entry points for tracing (the different `pub(crate)` functions) you'll see that they are infallible -- in particular they will not return an `Option<..>`. 

Instead, what they do is call into the internal fallible API that returns `Option<..>` and if a `None` bubbles all the way back up they will emit a trace error.